crates/matrix-sdk-sqlite/src/utils.rs

```diff
@@ -290,6 +304,48 @@ pub(crate) trait SqliteKeyValueStoreAsyncConnExt: SqliteAsyncConnExt {
 
     /// Store the given value for the given key.
     async fn set_kv(&self, key: &str, value: Vec<u8>) -> rusqlite::Result<()>;
+
+    /// Get the version of the database.
+    async fn db_version(&self) -> Result<u8, OpenStoreError> {
+        let kv_exists = self.kv_table_exists().await.map_err(OpenStoreError::LoadVersion)?;
+
+        if kv_exists {
+            match self.get_kv("version").await.map_err(OpenStoreError::LoadVersion)?.as_deref() {
```

Comment:
Do we need to check if the table exists before running the query? If the query fails, can't we assume it's because the table doesn't exist (maybe by inferring the `Error` type?), and return `Ok(0)` in this case?

The idea behind this proposal is to limit the number of queries. I am aware you've mostly moved code here, so feel free to ignore my comments if you believe it's too much or too risky changes.

Reply:
So the issue here is that SQLite returns a generic `ErrorCode::Unknown` in this case. The only way to clearly identify that the error is a missing table would be to match the beginning of the error message, which doesn't sound great.

Reply:
Indeed. Let's not change that then.